packages/material-ui/src/MobileStepper/MobileStepper.js

```diff
@@ -1,47 +1,100 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { integerPropType } from '@material-ui/utils';
-import withStyles from '../styles/withStyles';
+import { integerPropType, deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import Paper from '../Paper';
 import capitalize from '../utils/capitalize';
 import LinearProgress from '../LinearProgress';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    display: 'flex',
-    flexDirection: 'row',
-    justifyContent: 'space-between',
-    alignItems: 'center',
-    background: theme.palette.background.default,
-    padding: 8,
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import mobileStepperClasses, { getMobileStepperUtilityClass } from './mobileStepperClasses';
+
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(
+    {
+      ...styles[`position${capitalize(styleProps.position)}`],
+      [`& .${mobileStepperClasses.dots}`]: styles.dots,
+      [`& .${mobileStepperClasses.dot}`]: {
+        ...styles.dot,
+        ...(styleProps.dotActive && styles.dotActive),
+      },
```

Comment:
```suggestion
      [`& .${mobileStepperClasses.dot}`]: {
        ...styles.dot,
        ...(styleProps.dotActive && styles.dotActive),
      },
      [`& .${mobileStepperClasses.dotActive}`]: {
        ...styleProps.dotActive,
      },
```

we have to change it to this, as the prop is dynamic per dot.